src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+    public shape = 'triangle',
+  ) {
+    if (this.sideA <= 0 || this.sideB <= 0 || this.sideC <= 0) {
+      throw new Error(`Sides must be positive numbers`);
+    }
+
+    if (
+      sideA + sideB <= sideC ||
+      sideA + sideC <= sideB ||
+      sideB + sideC <= sideA
```

Comment:
The variables `sideA`, `sideB`, and `sideC` should be prefixed with `this.` to refer to the instance variables. Change `sideA` to `this.sideA`, `sideB` to `this.sideB`, and `sideC` to `this.sideC`.